storage/remote/otlptranslator/prometheusremotewrite/combined_appender.go

```diff
@@ -0,0 +1,221 @@
+// Copyright The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO(krajorama): rename this package to otlpappender or similar, as it is
+// not specific to Prometheus remote write anymore.
+// Note otlptranslator is already used by prometheus/otlptranslator repo.
+package prometheusremotewrite
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/metadata"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// CombinedAppender is similar to storage.Appender, but combines updates to
+// metadata, created timestamps, exemplars and samples into a single call.
+type CombinedAppender interface {
+	// AppendSample appends a sample and related exemplars, metadata, and
+	// created timestamp to the storage.
+	AppendSample(metricFamilyName string, ls labels.Labels, meta metadata.Metadata, ct, t int64, v float64, es []exemplar.Exemplar) error
+	// AppendHistogram appends a histogram and related exemplars, metadata, and
+	// created timestamp to the storage.
+	AppendHistogram(metricFamilyName string, ls labels.Labels, meta metadata.Metadata, ct, t int64, h *histogram.Histogram, es []exemplar.Exemplar) error
```

Comment:
[Nit] Is there a reason why `ct` comes before `t` in these method signatures, considering the ordering is the other way around for `HistogramAppender.AppendHistogramCTZeroSample` and `CreatedTimestampAppender.AppendCTZeroSample`? I fear it could be a source of confusion.

Reply:
@bwplotka requested it through https://github.com/prometheus/prometheus/pull/17104 , saying "I did swap t, ct with ct, t because ct has to be before, conceptually.".

Reply:
OK, thanks for the context.